planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -336,8 +336,19 @@ class PlanningSceneMonitor : private boost::noncopyable
   /** @brief This function is called every time there is a change to the planning scene */
   void triggerSceneUpdateEvent(SceneUpdateType update_type);
 
-  /** \brief Wait until all pending scene updates with timestamps < t are incorporated */
-  void syncSceneUpdates(const ros::Time &t = ros::Time::now());
+  /** \brief Wait for robot state to become more recent than t.
+   *
+   * If there is no state monitor active, there will be no scene updates.
+   * Hence, you can specify a timeout to wait for those updates. Default is 1s.
+   */
+  bool waitForCurrentRobotState(const ros::Time &t, double wait_time = 1.);
+
+  /** \brief Wait current robot state and wait for all pending scene updates to be processed.
+   *
+   * Additionally to waitForCurrentRobotState() this processes also all already received scene updates
+   * as long as wait_time is not exceeded.
+   */
+  bool syncSceneUpdates(const ros::Time &t = ros::Time::now(), double wait_time = 1.);
```

Comment:
If scene updates and planning requests are triggered from the very same program, the planning request is definitely processed after the update requests, isn't it?.
Thus it makes sense to have that function in place to ensure that all previous updates are considered for planning.

Of course, @v4hn is right, if another source for scene updates is involved. However, in that case the program sending the planning request cannot sync these updates beforehand as well.